File: TrabalhoC++/Avaliacao.cpp
#include "Avaliacao.hpp"

Avaliacao::Avaliacao(const std::string &disciplina, const std::string &nome, double peso, std::time_t data)
    : disciplinaKey(disciplina), nome(nome), peso(peso), data(data) {}

std::string Avaliacao::getDisciplinaKey() const
{
    return disciplinaKey;
}

std::string Avaliacao::getNome() const
{
    return nome;
}

double Avaliacao::getPeso() const
{
    return peso;
}

std::time_t Avaliacao::getData() const
{
    return data;
}

Avaliacao::Valores_WriteValueAvaliacaoAluno::Valores_WriteValueAvaliacaoAluno(double total, double qtd,
                                                                              double prova_final)
    : total(total), qtd(qtd), prova_final(prova_final) {}

double Avaliacao::Valores_WriteValueAvaliacaoAluno::getTotal() const
{
    return total;
}

double Avaliacao::Valores_WriteValueAvaliacaoAluno::getQtd() const
{
    return qtd;
}

double Avaliacao::Valores_WriteValueAvaliacaoAluno::getProva_final() const
{
    return prova_final;
}
/*

void Avaliacao::TratamentoExcecoes(std::string *dados, AlunoMap &alunos, AlunoMap &mapaAlunos,
                                   std::string &codigo, Avaliacao *avaliacao, double nota) {}
                                   */

File: testeTrabalhoJava/TrabalhoParteFinal/src/Avaliacao.java
package src;

import java.io.FileWriter;
import java.io.Serializable;
import java.util.Date;
import java.util.Map;

public abstract class Avaliacao implements Serializable {
    private String disciplinaKey;
    private String nome;
    private double peso;
    private Date data;

    public Avaliacao(String disciplina, String nome, double peso, Date data) {
        this.disciplinaKey = disciplina;
        this.nome = nome;
        this.peso = peso;
        this.data = data;
    }

    public String getDisciplinaKey() {
        return disciplinaKey;
    }

    public String getNome() {
        return nome;
    }

    public double getPeso() {
        return peso;
    }

    public Date getData() {
        return data;
    }

    public abstract void ModificaMapasNotas(Map<String, Integer> qtdNotas, Map<String, Double> totalNotas,
            Avaliacao avaliacao, String key_a, double value_NA);

    public abstract void WriteKeyAvaliacao(FileWriter writer, Avaliacao avaliacao, String key_avaliacao_aluno)
            throws Excecao;

    public abstract Valores_WriteValueAvaliacaoAluno WriteValueAvaliacaoAluno(FileWriter writer, Avaliacao avaliacao,
            double value_avaliacao_aluno)
            throws Excecao;

    public class Valores_WriteValueAvaliacaoAluno {
        private double total;
        private double qtd;
        private double prova_final;

        public Valores_WriteValueAvaliacaoAluno(double total, double qtd, double prova_final) {
            this.total = total;
            this.qtd = qtd;
            this.prova_final = prova_final;
        }

        public double getTotal() {
            return total;
        }

        public double getQtd() {
            return qtd;
        }

        public double getProva_final() {
            return prova_final;
        }
    }

    public abstract Avaliacao.Valores_WriteValueAvaliacaoAluno CalculaMediasAluno(Avaliacao avaliacao, double value_np);

    public abstract void TratamentoExcecoes(String[] dados, Map<Integer, Aluno> alunos,
            AlunoMap mapaAlunos,
            String codigo,
            Avaliacao avaliacao,
            double nota) throws Excecao;

}
